### ISSUE TITLE
Menu item "Photos" doesn't show any photos

### ISSUE THREAD

**Issue by SpamReceiver** [nextcloud/android#5893]
### Steps to reproduce
1. Photos stored on Nextcloud server, in own folders and in shared folders
2. Open Nextcloud Androud app, select "Photos" in menu.
3. Spinning wheel appears and after a few seconds a message is displayed in the otherwise empty view: "Keine Fotos. Laden Sie Fotos hoch older aktivieren Sie den Sofortupload."

### Expected behaviour
- Tell us what should happen
The photos stored on the server should be displayed.
This function worked some time ago. I can't explicitly confirm it, but it seems to have become broken when the "Gallery" app of the server was replaced with the "Photos" app (on the server).

### Actual behaviour
- Tell us what happens
No photos are displayed. 

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

**Didn't try.**

### Environment data
Android version: 9

Device model:  Galaxy S8

Stock or customized system: stock

Nextcloud app version: latest

Nextcloud server version: latest

Reverse proxy: no

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
>     * This function worked some time ago. I can't explicitly confirm it, but it seems to have become broken when the "Gallery" app of the server was replaced with the "Photos" app (on the server).

This should not be the case, as we use a "search" internally, so it is independent from any app.



> Didn't try.

Please try it.
I tested in on my local test system and there it it working just fine.
It loads 50 images and when you reach bottom, it lost next 50.

Maybe it is just timing out…?

**Comment by SpamReceiver** [comment]
> 
> 

> 
> > Didn't try.
> 
> Please try it.
> I tested in on my local test system and there it it working just fine.
> It loads 50 images and when you reach bottom, it lost next 50.
> 
> Maybe it is just timing out…?

Now, I have tested the app with try.nextcloud.com and the photos there are displayed in the app.
However, its only <10 photos, but I have >60,000 on my server.

**Comment by tobiasKaminsky** [comment]
Then I suspect that the search just takes way too long.
Can you open up an issue on server?
We cannot do anything, other than relying on what server gives us.

**Comment by SpamReceiver** [comment]
I'm not happy that you have closed the issue.
Can you please test with a server which hosts a lot of photos to see if it's really a problem of server response?
In the app, the spinning wheel spins for a few senconds before "No photos" is displayed" which is absolutely not correct.

**Comment by tobiasKaminsky** [comment]
I tested it with my server, which has >50k of images, but it is a quite fast server.
It really depends on server, so client side cannot do anything here.

**Comment by SpamReceiver** [comment]
So, how can I see in the client whether it receives any response from the server which tells "No photos available"?

**Comment by tobiasKaminsky** [comment]
Via logcat: https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat

**Comment by SpamReceiver** [comment]
Hello Tobias!

The issue remains. And it should not have been closed. It seems I'm not allowed to reopen it.

Why not close: There are lots of photos on the server and the app tells there are none. This is simply wrong.

I managed to install ADB on my computer and to call the command written on the page you had linked. It produces loads of output, so I need more information on how I would retrieve the relevant messages.

**Comment by SpamReceiver** [comment]
I played around a bit. Maybe this output is useful:

``
05-17 13:18:31.573  5429  5675 I InputDispatcher: Delivering touch to (2357): action: 0x0, displayId: 0, 'b4031bc', toolType: 1
05-17 13:18:31.573  2357  2357 D ViewRootImpl@8101b5f[FileDisplayActivity]: ViewPostIme pointer 0
05-17 13:18:31.854  5429  5675 I clp-JNI : Add pinfo.  (channel) b4031bc com.nextcloud.client/com.owncloud.android.ui.activity.FileDisplayAct
ivity (server) (pkg) AppWindowToken{fcb356e token=Token{22db4e9 ActivityRecord{7f2e970 u0 com.nextcloud.client/com.owncloud.android.ui.activi
ty.FileDisplayActivity t19504}}} 2357 10314 (action) 1
05-17 13:18:31.854  5429  5675 I InputDispatcher: Delivering touch to (2357): action: 0x1, displayId: 0, 'b4031bc', toolType: 1
05-17 13:18:31.856  2357  2357 D ViewRootImpl@8101b5f[FileDisplayActivity]: ViewPostIme pointer 1
05-17 13:18:32.096  2357  4833 D OwnCloudClient #0: REQUEST OPTIONS /nextcloud/remote.php/dav
05-17 13:18:32.098  2357  4833 D AdvancedSslSocketFactory: Creating SSL Socket with remote server.fritz.box:443, local null:0, params: org.ap
ache.commons.httpclient.params.HttpConnectionParams@90092f9
05-17 13:18:32.098  2357  4833 D AdvancedSslSocketFactory:  ... with connection timeout 60000 and socket timeout 60000
05-17 13:18:32.111  2357  4833 I ServerNameIndicator: SSLSocket implementation: com.google.android.gms.org.conscrypt.Java8FileDescriptorSocke
t
05-17 13:18:32.111  2357  4833 I ServerNameIndicator: SNI done, hostname: server.fritz.box
05-17 13:18:32.130  2357  2357 W extcloud.clien: Accessing hidden field Landroid/widget/Editor;->mDrawableForCursor:Landroid/graphics/drawabl
e/Drawable; (dark greylist, reflection)
05-17 13:18:32.142  2357  2357 W extcloud.clien: Accessing hidden field Landroid/widget/Editor;->mDrawableForCursor:Landroid/graphics/drawabl
e/Drawable; (dark greylist, reflection)
05-17 13:18:32.178  2357  4833 D OwnCloudClient #0: REQUEST SEARCH /nextcloud/remote.php/dav
05-17 13:18:32.180  2357  4833 D AdvancedSslSocketFactory: Creating SSL Socket with remote server.fritz.box:443, local null:0, params: org.ap
ache.commons.httpclient.params.HttpConnectionParams@d366c66
05-17 13:18:32.180  2357  4833 D AdvancedSslSocketFactory:  ... with connection timeout 60000 and socket timeout 60000
05-17 13:18:32.180  2357  4833 I ServerNameIndicator: SNI done, hostname: server.fritz.box
05-17 13:18:43.591  2357  4833 V NativeCrypto: SSL shutdown failed: ssl=0x7071fc41c8: I/O error during system call, Broken pipe

``

**Comment by BernhardGehl** [comment]
Any updates on this one? We seem to have the same issue here.
So I understand this is not a client but a server issue. What could we tell the server people so they understand the problem?

**Comment by SpamReceiver** [comment]
No updates. The issue still exists.

In my opinion, the client needs a correction as it displays a wrong status "No photos".

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky could we maybe distinguish this? My guess is that we run into the 1 minuter timeout so we could rather state that we couldn't retrieve the images in a acceptable time frame, no?

**Comment by SpamReceiver** [comment]
> @tobiasKaminsky could we maybe distinguish this? My guess is that we run into the 1 minuter timeout so we could rather state that we couldn't retrieve the images in a acceptable time frame, no?

I can see the timeout value of 60000 ms in the log I posted on May 17, but the client displays the "No photos" messages after less than 10 seconds.

The log indicates an SSL problem. Could you please check if the issue is related to the self-signed certificate of my (private) server?

**Comment by BernhardGehl** [comment]
I didn't have the opportunity to review my server logs, but 
* my server has a regular letsencrypt certificate
* I agree that there is a maximum of 10 seconds between clicking "Fotos" and seeing the error message.

Where do you see the SSL issue?

**Comment by SpamReceiver** [comment]
Please see my comment on May 17, 2020:

> 05-17 13:18:32.180 2357 4833 D AdvancedSslSocketFactory: Creating SSL Socket with remote server.fritz.box:443, local null:0, params: org.ap ache.commons.httpclient.params.HttpConnectionParams@d366c66 
> 
> 05-17 13:18:32.180 2357 4833 D AdvancedSslSocketFactory: ... with connection timeout 60000 and socket timeout 60000 
>
> 05-17 13:18:**_32_**.180 2357 4833 I ServerNameIndicator: SNI done, hostname: server.fritz.box 
> 
> 05-17 13:18:**_43_**.591 2357 4833 V **_NativeCrypto: SSL shutdown failed_**: ssl=0x7071fc41c8: I/O error during system call, Broken pipe`

**Comment by tobiasKaminsky** [comment]
@SpamReceiver can you test this also with another device/android version?

**Comment by SpamReceiver** [comment]
> @SpamReceiver can you test this also with another device/android version?

Sorry, no, I have only this device.

**Comment by SpamReceiver** [comment]
> @tobiasKaminsky I found some time to look at the issue:
> a) The issue was reproducible on...
>     * a Samsung Galaxy S10e Android 10 Stock ROM
>     * a Samsung Galaxy A40 Android 10 Stock ROM
>     * a Samsung Galaxy A20e Android 10 Stock ROM
>     * a Samsung Galaxy A3 Android 9 Stock ROM
>     * a Samsung Galaxy S4 Mini Android 8 LineageOS
>       (Sorry, currently we seem to be a quite Samsung-heavy family...)
>       --> On all phones, the "no Photos" message came after ~ 10 seconds.

Mine is a Samsung Galaxy S8 (SM-G950F) with Android 9.

> b) Since I was wondering if @SpamReceiver really had the same issue as me, I had a look at the logs of the S10e and found some similarities. The NativeCrypto ... Broken pipe error is there, but right before it, there is another line that looks interesting:
> 
> > 10-29	 15:13:16.561	23392	23445	D	OwnCloudClient #0: REQUEST SEARCH /remote.php/dav
> > 10-29	 15:13:16.561	23392	23445	V	NativeCrypto: SSL shutdown failed: ssl=0x71d4015548: I/O error during system call, Broken pipe
> > --> So the "Broken pipe" happens immediately after/in the search request?
> > There are some more lines with "OwnCloudClient #0", but they don't result in an errror message eg.:
> > 10-29	 15:13:16.427	23392	23445	D	OwnCloudClient #0: REQUEST OPTIONS /remote.php/dav
> > 10-29	 15:13:16.437	23392	23392	V	FileDisplayActivity: onResume() end
> > (Also there is some extensive chatter from ThemeUtils about some errors - but I don't think that's related. But who am I to tell...)

The `OwnCloudClient #0`message is in my log, too, but ~10 sec before the SSL shutdown.

> So looking at it, it I'd guess this to be a server-handset communication issue against some specific server configuration. Could it be that NextCloud itself is not really to blame but that there is a too restrictive rule set in the nginx reverse proxy?

My Nextcloud server runs on Apache2.

**Comment by tobiasKaminsky** [comment]
> Could it be that NextCloud itself is not really to blame but that there is a too restrictive rule set in the nginx reverse proxy?

I am unsure, as we use the same encryption/client settings for each and every call.
So I wonder why this only crashes on REQUEST SEARCH /remote.php/dav

@nextcloud/server-triage is /dav/ endpoint somehow different than the old "webdav" point in this regard?

**Comment by skjnldsv** [comment]
As far as I know that is just a redirection, should be the same.

**Comment by BernhardGehl** [comment]
So I did a bit of experimenting in my spare time... What I found out:
* When I start photos query, the load on the server (the mysqld container and an apache2 process in the nextcloud container) goes up with each process consuming ~ 50..90 % of a CPU. 
* Next I noticed, that the apache2 process persists in memory and CPU consumption even after the "Show-me-my-Photos" request fails on the phone leaving it as a Zombie process. (Actually that was an interesting find, because I had been wondering where these Zombies came from a few days back when the server was overrun with them.)
* Looking at `docker stats` the zombie keeps pulling in data via net IO (mariaDB probably, I wasn't sure how to check) while writing out about 0.1 times the input.
* The mariaDB container keeps running at ~ 50% CPU with only the zombie active.

Another observation is, that after increasing the memory ceiling of my nextcloud in the php.ini to 1 GB, the time before the phone fails increased a lot (though not quite to the 1 min timeout more ~ 30 s-ish). Increasing the ceiling to 2 GB didn't change anything. (The zombie's memory footprint looks kind of stable around 1 GB, maybe there's another limiting process built in.)

The Nextcloud error log doesn't show anything related (only the "Offset error" from this known bug: https://github.com/nextcloud/server/pull/23606 which probably isn't related.)

@tobiasKaminsky Could that help in pinpointing the source of the issue?

@SpamReceiver Could you check if you can replicate the behaviour observed on my system?

**Comment by tobiasKaminsky** [comment]
Thanks for your detailed check @BernhardGehl 
@nextcloud/server-triage this sounds like a bug on server: any webdav search clients provide should not lead to zombies or too much cpu…

**Comment by skjnldsv** [comment]
> @nextcloud/server-triage this sounds like a bug on server: any webdav search clients provide should not lead to zombies or too much cpu…

Well, search require the full filesystem setup. Meaning mounting all existing shares. The more you got, the more complex it is.
That's why we're not showing shares on Photos.

**Comment by tobiasKaminsky** [comment]
Sure, but if in case of @BernhardGehl one search leads to 50% cpu with zombie process, then I fear to think about 5 requests.
So this is a bug.
And if server knows about a problematic webdav search, it needs to be handled on server, as we cannot tell all clients / scripts out there to be careful with their searches as it might otherwise happen that the server dies…

**Comment by BernhardGehl** [comment]
I'd like to help but I have difficulty grasping the technical details:
@skjnldsv what does "mounting all existing shares" imply? Shares like external filesystem mounts - or folders shared by other users? Does this include __groupfolder from the Group Folders app? (I'm using that one and there's a big family photos album on it...)

Generally: Is there anything I can do to pin down the issue in more detail? How can I find out what the undead apaches are doing with all that CPU time?

**Comment by skjnldsv** [comment]
@BernhardGehl thanks for asking :)
Each share on nextcloud is a dedicated mount. Because users can move around the shares, it means to set up the full filesystem, we need to retrieve the list of all the user shares and their respective mounts and build from it. Which makes the whole searching very complicated as it would be like search in as many filesystems as you have shares (I'm simplifying it)

**Comment by BernhardGehl** [comment]
> @BernhardGehl thanks for asking :)
> Each share on nextcloud is a dedicated mount. Because users can move around the shares, it means to set up the full filesystem, we need to retrieve the list of all the user shares and their respective mounts and build from it. Which makes the whole searching very complicated as it would be like search in as many filesystems as you have shares (I'm simplifying it)

Would it be possible to trigger the same search that is requested from the Android app from a linux console from the Nextcloud Web interface? Like `SELECT * FROM gimmeallyourimages WHERE myuser IN hasaccess`?

**Comment by BernhardGehl** [comment]
OT: (btw. the ability to move around shares is a great feature when you get shares with unspeakable names and structure from your kids)

**Comment by SpamReceiver** [comment]
> So I did a bit of experimenting in my spare time... What I found out:
> 
>     * When I start photos query, the load on the server (the mysqld container and an apache2 process in the nextcloud container) goes up with each process consuming ~ 50..90 % of a CPU.

I don't use Docker. Apache2 and MariaDB are running on the host.

>     * Next I noticed, that the apache2 process persists in memory and CPU consumption even after the "Show-me-my-Photos" request fails on the phone leaving it as a Zombie process. (Actually that was an interesting find, because I had been wondering where these Zombies came from a few days back when the server was overrun with them.)
> 
>     * Looking at `docker stats` the zombie keeps pulling in data via net IO (mariaDB probably, I wasn't sure how to check) while writing out about 0.1 times the input.
> 
>     * The mariaDB container keeps running at ~ 50% CPU with only the zombie active.

Can you please explain how I identify zombies?

> Another observation is, that after increasing the memory ceiling of my nextcloud in the php.ini to 1 GB, the time before the phone fails increased a lot (though not quite to the 1 min timeout more ~ 30 s-ish). Increasing the ceiling to 2 GB didn't change anything. (The zombie's memory footprint looks kind of stable around 1 GB, maybe there's another limiting process built in.)

Do you want me to experiment with the PHP settings, too?

> The Nextcloud error log doesn't show anything related (only the "Offset error" from this known bug: [nextcloud/server#23606](https://github.com/nextcloud/server/pull/23606) which probably isn't related.)
> 
> @tobiasKaminsky Could that help in pinpointing the source of the issue?
> 
> @SpamReceiver Could you check if you can replicate the behaviour observed on my system?

I'll try to... see above

**Comment by BernhardGehl** [comment]
> I don't use Docker. Apache2 and MariaDB are running on the host

Good, do that would be a difference. :-)
 
> Can you please explain how I identify zombies?

Well, undead jokes aside, it was easy for me because there was exactly one apache2 process in my "top" output that popped up when I hit the Photos button in the app. It turned zombie, when the search threw it's "no photos" screen and the apache2 process kept using up 50+% of CPU. A second search would start another apache2.

Since my Nextcloud is a family install and all other users were in bed and asleep, I could be quite sure that the process originated from the Android app's request. To be on the safe side, the behavior could be replicated after restarting the Nextcloud container.

Okay from a definition side?

> Do you want me to experiment with the PHP settings, too?

You can do that if you're comfortable but I'm most interested in the above mentioned zombie behavior that was independent from any memory ceiling I had set.

@SpamReceiver: Could you check?

**Comment by tobiasKaminsky** [comment]
@nextcloud/server-triage this seems to affect more and more people…

**Comment by vitormattos** [comment]
I have the same problem, I believe it is some problem with the server's `/remote.php/dav` route.

Pictures don't appear to me either. When I pull down the Photos section in the app, this appears in the `logcat`:

```logcat
12-06 16: 41: 08.570 21641 21661 D OwnCloudClient # 0: REQUEST OPTIONS /remote.php/dav
12-06 16: 41: 08.591 21641 21641 W extcloud.clien: Accessing hidden field Landroid / widget / Editor; -> mDrawableForCursor: Landroid / graphics / drawable / Drawable; (dark greylist, reflection)
12-06 16: 41: 08.599 21641 21641 W extcloud.clien: Accessing hidden field Landroid / widget / Editor; -> mDrawableForCursor: Landroid / graphics / drawable / Drawable; (dark greylist, reflection)
12-06 16: 41: 08.840 21641 21661 D RemoteOperationResult: RemoteOperationResult has processed UNHANDLED_HTTP_CODE: 204
```

I believe that code 204 indicates the problem but I have not yet identified what causes response code 204

**Comment by skruehoetl** [comment]
The issue resolved itself with Nextcloud ~20.0.6.
Could be closed from my side.

**Comment by SpamReceiver** [comment]
> The issue resolved itself with Nextcloud ~20.0.6.
> Could be closed from my side.

Menu item "Photos" does no longer exist.
It has been replaced by "Media" which works for me.

I agree to close this issue.

**Comment by tobiasKaminsky** [comment]
Thanks for feedback and glad to hear that it works now :+1:
